test/language/expressions/assignment/destructuring/obj-prop-put-const-no-strict.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    The assignment target should obey `const` semantics outside of strict mode.
+es6id: 12.14.5.4
+flags: [noStrict]
+features: [const]
+---*/
+
+var value = { a: 2 };
+var result;
+const c = 1;
+
+result = { a: c } = value;
```

Comment:
Assignment to immutable strict binding throws a TypeError in non-strict (and strict) mode (8.1.1.1.5 SetMutableBinding, step 6). (`const` is always created as immutable strict binding, cf. 15.1.8 Runtime Semantics: GlobalDeclarationInstantiation, step 15.b.i.1).